code/modules/paperwork/stamps.dm

```diff
@@ -1,55 +1,46 @@
 /obj/item/stamp
 	name        = "rubber stamp"
 	desc        = "A rubber stamp for stamping important documents."
-	icon        = 'icons/obj/items/rubber_stamps.dmi'
-	icon_state  = "stamp-deckchief"
-	item_state  = "stamp"
-	throwforce  = 0
+	icon        = 'icons/obj/items/stamps/stamp_deckchief.dmi'
+	icon_state  = ICON_STATE_WORLD
+	throwforce  = 1
 	w_class     = ITEM_SIZE_TINY
 	throw_speed = 7
 	throw_range = 15
 	material    = /decl/material/solid/metal/steel
+	item_flags  = ITEM_FLAG_STAMP
 	matter      = list(
 		/decl/material/solid/organic/plastic = MATTER_AMOUNT_REINFORCEMENT,
 	)
 	attack_verb = list("stamped")
 
 /obj/item/stamp/denied
 	name = "\improper DENIED rubber stamp"
-	icon_state = "stamp-deny"
-	attack_verb = list("stamped", "denied")
+	icon = 'icons/obj/items/stamps/stamp_deny.dmi'
 
 /obj/item/stamp/clown
 	name = "clown's rubber stamp"
-	icon_state = "stamp-clown"
+	icon = 'icons/obj/items/stamps/stamp_clown.dmi'
 
 /obj/item/stamp/boss
 	name = "boss' rubber stamp"
-	icon_state = "stamp-boss"
+	icon = 'icons/obj/items/stamps/stamp_boss.dmi'
 
 /obj/item/stamp/boss/Initialize()
 	name = "[global.using_map.boss_name]'s' rubber stamp"
 	. = ..()
 
 // Syndicate stamp to forge documents.
 /obj/item/stamp/chameleon/attack_self(mob/user)
-
-	var/list/stamp_types = typesof(/obj/item/stamp) - src.type // Get all stamp types except our own
 	var/list/stamps = list()
-
 	// Generate them into a list
-	for(var/stamp_type in stamp_types)
-		var/obj/item/stamp/S = new stamp_type
-		stamps[capitalize(S.name)] = S
-
+	for(var/stamp_type in subtypesof(/obj/item/stamp))
```

Comment:
noooo. `typesof(/obj/item/stamp) - /obj/item/stamp/chameleon` is not equivalent to `subtypesof(/obj/item/stamp)`. this will exclude the deck chief stamp but include the chameleon stamp itself.
```suggestion
	for(var/stamp_type in typesof(/obj/item/stamp) - type) // exclude the chameleon stamp subtype
```